subprojects/logging/src/integTest/groovy/org/gradle/internal/logging/console/AbstractConsoleBuildPhaseFunctionalTest.groovy

```diff
@@ -439,4 +483,31 @@ abstract class AbstractConsoleBuildPhaseFunctionalTest extends AbstractConsoleGr
     String regexFor(String message) {
         /<.*> $message \[[\dms ]+]/
     }
+
+    void waitForFinish() {
+        result = gradle.waitForFinish()
+        outputDoesNotContain("More progress was logged than there should be")
+    }
+
+    String buildFinishedCall(String name) {
+        """
+            abstract class BuildFinishCall implements FlowAction<Parameters> {
+                interface Parameters extends FlowParameters {
+                    @Input
+                    Property<String> getBuildSuccess();
```

Comment:
💭 I have this property only here so it can depend on `buildWorkResult` and runs at the end of the build. I wonder if there is a better way to do so...

Reply:
The flow action needs to be connected to the provider so it'll trigger at the right point in time.

Reply:
Yeah, I understand. I still think this is quite some ceremony and wonder if there should be a more straightforward way.